drivers/can/can_mcp251xfd.c

```diff
@@ -30,7 +30,7 @@ LOG_MODULE_REGISTER(can_mcp251xfd, CONFIG_CAN_LOG_LEVEL);
 static void mcp251xfd_canframe_to_txobj(const struct can_frame *src, int mailbox_idx,
 					struct mcp251xfd_txobj *dst)
 {
-	memset(dst, 0, sizeof(*dst));
+	memset(dst, 0, offsetof(struct can_frame, data));
```

Comment:
Since the `MCP251XFD_RAM_ALIGNMENT` is 4 and you later round up to this when transmitting data on the SPI bus, but you only copy up to DLC number of bytes, you still risk leaking uninitialized memory by not zeroing it here.

Reply:
That aside, you likely meant to write `offsetof(struct mcp251xfd_txobj, data)`.

Reply:
fair enough, good catch, I'll just drop this part of the PR.